apps/cmp.c

```diff
@@ -1761,6 +1761,7 @@ static int setup_request_ctx(OSSL_CMP_CTX *ctx, ENGINE *engine)
     CMP_err("out of memory");
  exts_err:
     sk_X509_EXTENSION_pop_free(exts, X509_EXTENSION_free);
+    X509_REQ_free(csr);
```

Comment:
After this fixup you'll need to assign NULL to csr after the explicit free at line 1687. And I am a little bit suspicious about the exts - whether there is also some possible doublefree. It should be assigned NULL after the   OSSL_CMP_CTX_set0_reqExtensions().

Reply:
Ah, indeed there is a problem.The idea was that after
```
        OSSL_CMP_CTX_set0_reqExtensions(ctx, exts);
    }
    X509_REQ_free(csr);
```
in lines 1685-1687 no free() is called for `exts` and `csrl` within that function and that
the cleanup code starting with `oom:` (and `exts_err`) is relevant only for code before that.
Yet there is one `goto oom` later that violates my assumption.
So I've added respective assignments to `NULL`.
Thanks for the careful checking.